src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -1550,16 +1551,48 @@ public function get_test_https_status() {
 					__( 'HTTPS is already supported for your website.' )
 				);
 
-				$result['actions'] = sprintf(
-					'<p><a href="%s">%s</a></p>',
-					esc_url( admin_url( 'options-general.php' ) ),
-					__( 'Update your site addresses' )
-				);
+				if ( defined( 'WP_HOME' ) || defined( 'WP_SITEURL' ) ) {
+					$result['description'] .= sprintf(
+						'<p>%s</p>',
+						__( 'However, your WordPress Address is currently controlled by a PHP constant and therefore cannot be updated. You need to edit your <code>wp-config.php</code> and remove or update the definitions of <code>WP_HOME</code> and <code>WP_SITEURL</code>.' )
```

Comment:
Should the wp-config and constant references be pulled out into a sprintf placeholders? Looking at `wp-admin/includes/network.php` for example.